core/chains/evm/gas/models.go

```diff
@@ -337,6 +343,47 @@ func (e *evmFeeEstimator) BumpFee(ctx context.Context, originalFee EvmFee, feeLi
 	return
 }
 
+func (e *evmFeeEstimator) estimateFeeLimit(ctx context.Context, feeLimit uint64, calldata []byte, toAddress *common.Address) (estimatedFeeLimit uint64, err error) {
+	// Use provided fee limit by default if EstimateGasLimit is disabled
+	if !e.geCfg.EstimateGasLimit() {
+		return commonfee.ApplyMultiplier(feeLimit, e.geCfg.LimitMultiplier())
+	}
+	// Create call msg for gas limit estimation
+	// Skip setting Gas to avoid capping the results of the estimation
+	callMsg := ethereum.CallMsg{
+		To:   toAddress,
+		Data: calldata,
+	}
+	estimatedGas, estimateErr := e.ethClient.EstimateGas(ctx, callMsg)
+	if estimateErr != nil {
```

Comment:
In the future, we could perhaps add a retry.

Reply:
I think we technically already do. This would surface as a gas estimator error which are retryable [here](https://github.com/smartcontractkit/chainlink/blob/d7645dd91c6d644cdebcd3590a5fd4da6cb260bc/core/chains/evm/txmgr/attempts.go#L63)

Reply:
@dimriou do you mean eth client level retry or tx attempt retry ?